daemon/daemon.go

```diff
@@ -193,15 +183,15 @@ configRetry:
 			continue configRetry
 		}
 		// Parse and merge the local config.
-		configParams.UpdateFrom(envConfig, config.EnvironmentVariable)
-		if configParams.Err != nil {
+		_, err = configParams.UpdateFrom(envConfig, config.EnvironmentVariable)
+		if err != nil || configParams.Err != nil {
```

Comment:
This code was using the design pattern used by the `bufio.Scanner`, where it collects its errors so the calling code doesn't need so much boilerplate. After your changes, we now have the error collection code and the boilerplate, which makes this code worse.

Think we should do one of these things:

- Stick with the pattern and move it closer to the Scanner API so that the linter doesn't complain (i.e. stop returning the error from each `Update...()` call.  `UpdateFrom(...) bool` and `Err() error`
- Ditch `configParams.Err` and just return the error and live with the boilerplate.

Reply:
The second option would allow us to have more specific error message, which is always nice to have when debugging something like this. So I'm going with that one, but let me know if you prefer otherwise.